### ISSUE TITLE
Added source as an input variable for Rshiny differentiation.

### ISSUE THREAD

**Issue by PatrickvanZalm** [talgalili/heatmaply#295]
We noticed that when there is more than 1 heatmaply in Rshiny, plotly::event_data would collect clicks for both, making differentiating between plot 1 and plot 2 very difficult.

Plotly does have an input variable called source. I make a fork of this repo and added this input variable to the Heatmaply code. Interestingly, it was easiest to overwrite the (default) source by the input source. Just pushing it to the regular heatmap object before px or py are added through the layout() function would not lead to the desired result.

https://github.com/PatrickvanZalm/heatmaply

I tested it with the following code:

```r
devtools::load_all("/home/heatmaply")
library(shiny)
library(plotly)


ui <- fluidPage(
  titlePanel("Old Faithful Geyser Data"),
  mainPanel(
    plotlyOutput(outputId = 'heatmap')
  )
)

# Define server logic required to draw a histogram
server <- function(input, output) {
  
  data <- rnorm(100) %>% matrix(., 10, 10)
  hm <- heatmaply(data, source = "B")
  
  output$heatmap <- renderPlotly({
    hm
  })
  
  observe({
    click_event <- event_data("plotly_click", source = "B")
    print(click_event)
  })
}

# Run the application 
shinyApp(ui = ui, server = server)
```

If it looks good on your end, I can initiate the pull request.

**Comment by alanocallaghan** [comment]
Can you link me some documentation on the source feature?

**Comment by PatrickvanZalm** [comment]
Hi,

Sorry, I have not done a PR before so might need a bit of guidance here.

As described in the plot_ly fuction description (https://rdrr.io/cran/plotly/src/R/plotly.R) the 'source' variable helps identify the right plot in the event_data() to retrieve events to a specific plot, as there can be more than one.

In the forked repo, all I did was overwrite the source string after the heatmap, dendogram have been added together. I tried doing it on only the heatmap, but doing so I could not capture the event correctly.
